pkg/upstream/cluster/cluster_manager.go

```diff
@@ -61,11 +61,64 @@ const globalTLSMetrics = "global"
 
 // types.ClusterManager
 type clusterManager struct {
-	clustersMap      sync.Map
-	protocolConnPool sync.Map // protocolname: { address : connpool }
-	tlsMetrics       *mtls.TLSStats
-	tlsMng           atomic.Value // store types.TLSClientContextManager
-	mux              sync.Mutex
+	clustersMap       sync.Map
+	clusterPoolEnable bool // TODO support modify clusterPoolEnable at runtime
+	protocolConnPool  *protocolConnectionPool
+	tlsMetrics        *mtls.TLSStats
+	tlsMng            atomic.Value // store types.TLSClientContextManager
+	mux               sync.Mutex
+}
+
+type protocolConnectionPool struct {
```

Comment:
I feel there is still some redundancy. First of all, `protocolConnPool` can be renamed to `connPool`,  and `load` return the connpool directly that user don't need to sense whether clusterpool.

Reply:
I've simplified it.